Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/storage/TestCertificateContainer.kt

```diff
@@ -132,8 +134,13 @@ data class TestCertificateContainer(
             override val headerExpiresAt: Instant
                 get() = header.expiresAt
 
-            override val qrCode: QrCodeString
-                get() = data.testCertificateQrCode!!
+            override val qrCodeToDisplay: CoilQrCode = when (getState()) {
+                State.Invalid -> when (userLocale.language) {
+                    Locale.GERMAN.language -> State.Invalid.URL_INVALID_SIGNATURE_DE
+                    else -> State.Invalid.URL_INVALID_SIGNATURE_EN
+                }.let { CoilQrCode(it, QrCodeOptions(correctionLevel = ErrorCorrectionLevel.M)) }
+                else -> CoilQrCode(data.testCertificateQrCode!!)
+            }
```

Comment:
Magic happens here :mage_woman:

Reply:
can we move the magic snippet into a common magic place ?  😉